WordPress/src/main/java/org/wordpress/android/ui/sitecreation/domains/SiteCreationDomainSanitizer.kt

```diff
@@ -0,0 +1,24 @@
+package org.wordpress.android.ui.sitecreation.domains
+
+import java.util.Locale
+import javax.inject.Inject
+
+class SiteCreationDomainSanitizer
+@Inject constructor() {
+    /**
+     * Sanitizes a query by taking the value before it's first period if it's present,
+     * removes it's scheme and finally, removes any characters that aren't alphanumeric.
+     */
+    fun sanitizeDomainQuery(query: String) =
+            query.run {
+                val periodIndex = indexOf(".")
+                if (periodIndex > 0) {
```

Comment:
Just to double check: I noticed you use `>` without `=`. Is this intentional, because you know that if the query starts with `.`, the dot will get removed in `.replace("[^a-zA-Z0-9]".toRegex(), "")`?

SideNote: We might want to add a unit tests for this scenario. Wdyt?

Reply:
Yes, that was intentional. I thought if it begins with a `.` we should just remove it altogether. Especially since this check is ensuring that there will be characters present to perform the substring operation on. 

Yes, I was thinking to write a test for it but then I decided that the test case for the alphanumeric case covered it but it's something that can be done if we decide to change it's implementation.

Reply:
👍 Thanks! ;)